js/src/components/validation-errors/validation-errors.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Internal dependencies
+ */
+import './validation-errors.scss';
+
+/**
+ * Renders form validation error messages.
+ *
+ * @param {Object} props React props
+ * @param {string|string[]} [props.messages] Validation error message(s).
+ */
+export default function ValidationErrors( { messages: rawMessages } ) {
+	let messages = rawMessages;
+
+	if ( ! rawMessages?.length ) {
+		return null;
+	}
+
+	if ( ! Array.isArray( rawMessages ) ) {
+		messages = [ rawMessages ];
+	}
+
+	return (
+		<ul className="gla-validation-errors">
+			{ messages.map( ( message ) => (
+				<li key={ message }>{ message }</li>
+			) ) }
+		</ul>
+	);
+}
```

Comment:
This function was a bit hard for me to read, ideally, I would expect to only allow an array.

If that's not possible I think using another naming instead of `messages` for the local variable would make it easier to understand. At least for me, the fact that the local variable `messages` is the same as the argument `massages` an right after, we assigned it to `rawMessages` wich is the name overrides for `messages` made it hard to understand at first sight.

A potential way to do it might be:

```jsx
export default function ValidationErrors( { messages } ) {
        let messagesList = messages;  

        if ( ! messages?.length ) {
		return null;
	}
	
        if ( ! Array.isArray( messages ) ) {
		messagesList = [ messages ];
	}
	

	return (
		<ul className="gla-validation-errors">
			{ messagesList.map( ( message ) => (
				<li key={ message }>{ message }</li>
			) ) }
		</ul>
	);
}
```

Reply:
Renamed in https://github.com/woocommerce/google-listings-and-ads/pull/2019/commits/10df591df6bdf0c0bc8ac68e399776c7cab7544a.